src/prism.c

```diff
@@ -5836,8 +5836,13 @@ pm_string_node_create_unescaped(pm_parser_t *parser, const pm_token_t *opening,
     pm_string_node_t *node = PM_ALLOC_NODE(parser, pm_string_node_t);
     pm_node_flags_t flags = 0;
 
-    if (parser->frozen_string_literal) {
-        flags = PM_NODE_FLAG_STATIC_LITERAL | PM_STRING_FLAGS_FROZEN;
+    switch (parser->frozen_string_literal) {
+        case -1:
+            flags = PM_NODE_FLAG_STATIC_LITERAL | PM_STRING_FLAGS_MUTABLE;
```

Comment:
It shouldn't be seen as a static literal if it is mutable. This changes stuff like `["foo"]` which will `newarray`/`duparray` depending on the contents.